gto-support-kotlin-coroutines/src/test/kotlin/org/ccci/gto/android/common/kotlin/coroutines/ReadWriteMutexTest.kt

```diff
@@ -139,6 +141,31 @@ class ReadWriteMutexTest {
         mutex.read.unlock()
     }
 
+    @Test(timeout = 10000)
+    fun `GT-1423 readUnlock causes deadlock from runBlocking usage`() = runBlocking {
+        mutex.write.lock()
+
+        launch(Dispatchers.Unconfined) {
+            expect(1)
```

Comment:
Do these `expect` calls refer to the next line? e.g. Expect `mutex.read.lock()` to be called once?

Reply:
they count through the expected order of execution. the `expect()` and `finish()` methods can be seen at the bottom of the test.

I just updated the description describing what I'm trying to fix in case you hadn't seen it